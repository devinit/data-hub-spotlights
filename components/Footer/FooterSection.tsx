import React, { FunctionComponent } from 'react';
import { FooterSection as FooterSectionType, NavigationItem, SocialLink } from '../DefaultLayout';

export type FooterSectionProps = FooterSectionType & { primaryNavigation: NavigationItem[] };
interface SectionLinkProps {
  url: string;
  label?: string;
}

export const FooterSectionLink: FunctionComponent<SectionLinkProps> = ({ children, url, label }) =>
  <li><a href={ url }>{ label || children }</a></li>;

export const FooterSocialLink: FunctionComponent<SocialLink> = ({ social_platform, link_url, image_url }) =>
  <FooterSectionLink url={ link_url }>
    <img src={ image_url } alt={ social_platform }/>
  </FooterSectionLink>;

const FooterSection: FunctionComponent<FooterSectionProps> = props => {
  return (
    <div className="l-footer__col">
      <h4 className="footer__title">{ props.title }</h4>
      <ul className="footer__list">
        {
          props.primaryNavigation.map(({ full_url, title }, index) =>
            <FooterSectionLink key={ `${title} ${index}` } url={ full_url } label={ title }/>
          )
        }
        {
          props.section_links.map(({ link_url, page_url, label }, index) =>
            <FooterSectionLink key={ `${label} ${index}` } url={ link_url || page_url } label={ label }/>
          )
        }
      </ul>
      <ul className="footer__social">
        {
          props.social_links.map(({ link_url, social_platform, image_url }, index) =>
            <FooterSocialLink
              key={ `${social_platform} ${index}` }
              link_url={ link_url }
              image_url={ image_url }
              social_platform={ social_platform }
            />
          )
        }
      </ul>
    </div>
  );
};

export { FooterSection };
