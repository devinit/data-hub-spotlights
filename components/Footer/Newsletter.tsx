import React, { FunctionComponent } from 'react';
import { Newsletter as NewsletterType } from '../DefaultLayout';

type NewsletterProps = NewsletterType;

const Newsletter: FunctionComponent<NewsletterProps> = ({ caption, link_label, link_url }) => {
  return (
    <>
      <h4 className="footer__title">{ caption }</h4>
      <a href={ link_url } className="button">{ link_label }</a>
    </>
  );
};

export { Newsletter };
