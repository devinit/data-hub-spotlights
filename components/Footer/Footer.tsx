import React, { FunctionComponent } from 'react';
import { Footer as FooterData, NavigationItem } from '../DefaultLayout';
import { Newsletter } from './Newsletter';
import { FooterSection } from './FooterSection';

type FooterProps = Partial<FooterData> & { primaryNavigation: NavigationItem[] };

const Footer: FunctionComponent<FooterProps> = props => {
  const createMarkup = (html: string) => {
    return { __html: html };
  };

  return (
    <footer role="contentinfo" className="footer">
      <div className="row">
        <div className="l-footer">
          <div className="l-footer__col">
            {props.newsletters
              ? props.newsletters.map(letter => <Newsletter key={letter.caption} {...letter} />)
              : null}
          </div>
          {props.sections &&
            props.sections.map((section, index) => (
              <FooterSection key={index} {...section} primaryNavigation={props.primaryNavigation} />
            ))}
        </div>
        <div className="footer__meta">
          <div
            className="footer__meta-copy footer__meta-copy--major"
            dangerouslySetInnerHTML={createMarkup(props.footer_text_major || '')}
          />
          <div className="footer__meta-copy" dangerouslySetInnerHTML={createMarkup(props.footer_text || '')} />
        </div>
      </div>
    </footer>
  );
};

export { Footer };
