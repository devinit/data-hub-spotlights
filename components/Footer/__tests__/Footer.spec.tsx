/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { Footer } from '../Footer';

describe('Footer', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(
      <Footer/>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
