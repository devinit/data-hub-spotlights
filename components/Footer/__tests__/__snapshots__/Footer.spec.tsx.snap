// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  className="footer"
  role="contentinfo"
>
  <div
    className="row"
  >
    <div
      className="l-footer"
    >
      <div
        className="l-footer__col"
      >
        <h4
          className="footer__title"
        >
          Sign up for our newsletters
        </h4>
        <a
          className="button"
          href="http://localhost:3000/newsletter"
        >
          Sign up
        </a>
      </div>
      <div
        className="l-footer__col"
      >
        <h4
          className="footer__title"
        >
          More links
        </h4>
        <ul
          className="footer__list"
        >
          <li>
            <a
              href="https://localhost:3000/nav-1"
            >
              Nav 1
            </a>
          </li>
          <li>
            <a
              href="https://localhost:3000/nav-2"
            >
              Nav 2
            </a>
          </li>
          <li>
            <a
              href="http://localhost:3000/contact-us"
            >
              Contact us
            </a>
          </li>
        </ul>
        <ul
          className="footer__social"
        >
          <li>
            <a
              href="http://localhost:3000/contact-us"
            >
              <img
                alt="facebook"
                src="http://localhost:3000/assets/images/image.svg"
              />
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="footer__meta"
    >
      <div
        className="footer__meta-copy footer__meta-copy--major"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Text Major",
          }
        }
      />
      <div
        className="footer__meta-copy"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Text Minor",
          }
        }
      />
    </div>
  </div>
</footer>
`;
